app/assets/javascripts/discourse/app/models/post-stream.js

```diff
@@ -71,6 +75,13 @@ export default RestModel.extend({
       loadingFilter: false,
       stagingPost: false,
       timelineLookup: [],
+      topicSummary: {
```

Comment:
How do you feel about creating a separate `models/topic-summary.js` file to contain all of this stuff in a native class? The PostStream model is already pretty complicated, so splitting out might help with readability?

Then we could replace this line with something like:

```
topicSummary: new TopicSummary(topicId)
```

Reply:
Let's do that!